internal/internal_task_handlers.go

```diff
@@ -943,43 +873,97 @@ func isDecisionMatchEvent(d *s.Decision, e *s.HistoryEvent, strictMode bool) boo
 }
 
 func (wth *workflowTaskHandlerImpl) completeWorkflow(
-	isWorkflowCompleted bool,
-	completionResult []byte,
-	err error,
-) *s.Decision {
-	var decision *s.Decision
-	if canceledErr, ok := err.(*CanceledError); ok {
-		// Workflow cancelled
-		decision = createNewDecision(s.DecisionTypeCancelWorkflowExecution)
-		decision.CancelWorkflowExecutionDecisionAttributes = &s.CancelWorkflowExecutionDecisionAttributes{
-			Details: canceledErr.details,
+	eventHandler *workflowExecutionEventHandlerImpl,
+	task *s.PollForDecisionTaskResponse,
+	workflowContext *workflowExecutionContext,
+	decisions []*s.Decision) (completeRequest interface{}) {
+
+	if panicErr, ok := workflowContext.err.(*PanicError); ok {
+		// Workflow panic
+		wth.metricsScope.Counter(metrics.DecisionTaskPanicCounter).Inc(1)
+		wth.logger.Error("Workflow panic.",
+			zap.String("PanicError", panicErr.Error()),
+			zap.String("PanicStack", panicErr.StackTrace()))
+		failedCause := s.DecisionTaskFailedCauseWorkflowWorkerUnhandledFailure
+		_, details := getErrorDetails(panicErr)
+		completeRequest = &s.RespondDecisionTaskFailedRequest{
```

Comment:
I would change it to return to avoid all these elses. So code would look like:

if ... {
   return &s....
}
if ... {
   return &s...
}